### PR TITLE
US601302: Squashed all spring enabler related changes from branch US6…

### PR DESCRIPTION
**Pull request by pinpan**
…02301_onboarding-enablers-spring

#### Is there a related issue for this PR? 
Issue number:

### CODE REVIEW

File: docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler [hunk 1]
```diff
@@ -0,0 +1,502 @@
+# Onboarding a Spring Boot based REST API Service
```
Comment (taban03) [inline docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler]: Save as `.md` file and add `@ComponentScan(value = {
		"org.zowe.apiml.enable"})` mention

File: docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler [hunk 2]
```diff
@@ -0,0 +1,502 @@
...
+
+This guide is part of a series of guides to onboard a REST API service with the Zowe API Mediation Layer.
+As an API developer, you can onboard your REST API service built with the Spring Boot framework with the Zowe API Mediation Layer.
+
+**Note:** Before version 1.12, the API ML provided an integration enabler based on Spring Cloud Netflix components. From version 1.12 and later, the enabler has been replaced with new implimentation based on the Plain Java Enabler (PJE) that is not backwards compatiable with the previous enabler versions.
```
Comment (ilkinabdullayev) [inline docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler]: maybe link for Plain Java Enabler (PJE) ?

File: docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+- onboarding-enabler-spring-v2
+
+**Note:** The process of onboarding an API service is the same for both Spring Boot enabler versions.
+
+## Configuring your project
+
+Use either _Gradle_ or _Maven_ build automation systems to manage your project builds.
+
+**Note:** You can download the selected enabler artifact from the Giza Artifactory or  if you decide to build the API ML from source, you are required to publish the enabler artifact to your Artifactory.
+ Publish the enabler artifact by using the _Gradle_ tasks provided in the source code.
```
Comment (ilkinabdullayev) [inline docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler]: We can discuss around these points;
Do we need to mention which gradle publish artifact ?

File: docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler [hunk 4]
```diff
@@ -0,0 +1,502 @@
...
+
+<font color="red">TODO#FindOut what name will be SE published under</font>
+
+```gradle
+implementation "com.ca.mfaas.sdk:mfaas-integration-enabler-java:$zoweApimlVersion"
+implementation "com.ca.mfaas.sdk:common-service-core:$zoweApimlVersion"
+```
+**Note:** The published artifact from the Giza Artifactory also contains the enabler dependencies from other software packages. If you are using an Artifactory other than Giza, manually provide the following dependencies in your service `build.gradle` script:
+
+<font color="red">TODO#Check what other dependencies are required for SE</font>
```
Comment (ilkinabdullayev) [inline docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler]: todo

File: docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler (hunk 2 above)
Comment (pinpan) [reply docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler]: Link to the PJE documentation is provided at line 11 close to the beginning of the file.

File: docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler (hunk 3 above)
Comment (pinpan) [reply docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler]: The idea behind is, that this is an open source, so if the user decides to clone/fork the source code and make changes, they can, but then they need to publish to a artifactory to which they have write access.

File: docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler (hunk 1 above)
Comment (pinpan) [reply docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler]: Added md extension.

File: docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler [hunk 5]
```diff
@@ -0,0 +1,502 @@
...
+            }
+        }
+    }
+    ```
+4.  In the same `build.gradle` file, add the necessary dependencies for your service. If you use the _SpringEnabler_ from the Giza Artifactory, add the following code block to your `build.gradle` script:
+
+<font color="red">TODO#FindOut what name will be SE published under</font>
+
+```gradle
+implementation "com.ca.mfaas.sdk:mfaas-integration-enabler-java:$zoweApimlVersion"
```
Comment (taban03) [inline docs/extend/extend-apiml/api-mediation-onboarding-with-spring-boot-enabler]: Every `com.ca.*` is now changed to `org.zowe.apiml` so we need to update also this markdown.

**Pull request merged**